content/en/blog/attribute-autocomplete-enhancement/index.md

```diff
@@ -0,0 +1,306 @@
+      <constraints>
+    },
+    ...
+  }
+}
+```
+
+Here are the attribute type schemas for the three operators. Those schemas cover the most common situations.
+
+### Check if the attribute type is in a set
+
+In "Sentiment Analysis" operator, the attribute type of `attribute` must be string.
+
+```JSON
+{  
+  "attributeTypeRules": {  
+    "attribute": {  
+      "enum": ["string"]  
+    }  
+  }  
+}
+```
+
+### Check the equality of two attribute types
+
+In "Hash Join" operator, the attribute type of `buildAttributeName` (Left Input Attribute) must equal to that of `probeAttributeName` (Right Input Attribute).
+
+```JSON
+{
+  "attributeTypeRules":{
+    "buildAttributeName":{
+      "const":{
+        "$data":"probeAttributeName"
+      }
+    }
+  }
+}
+```
+
+*Note: the  `$data`  keyword is used to implement value comparison. Currently, it is the only way to check equality between values in JSON Schema. However, the  `$data`  keyword proposal is included not in the formal JSON Schema standard. See  [json-schema-org/json-schema-spec#51](https://github.com/json-schema-org/json-schema-spec/issues/51), and  [https://ajv.js.org/guide/combining-schemas.html#data-reference](https://ajv.js.org/guide/combining-schemas.html#data-reference).*
+
+### Attribute type dependent on other property's value
+
+In "Aggregate" operator, if  `aggFunction`'s value is "sum", "average", "min", or "max", then the attribute type of "attribute" can only be numeric (integer, long, double, or timestamp). Similarly, if  `aggFunction`'s value is "concat", then the attribute type of "attribute must be "string". If  `aggFunction`  has other values, like "count", then it has no constraint.
+
+All  `if-then`  in  the `allOf`  array must be satisfied. 
+
+```JSON
+{
+  "attributeTypeRules": {
+    "attribute": {
+      "allOf": [
+        {
+          "if": {
+            "aggFunction": {
+              "valEnum": ["sum", "average", "min", "max"]
+            }
+          },
+          "then": {
+            "enum": ["integer", "long", "double", "timestamp"]
+          }
+        },
+        {
+          "if": {
+            "aggFunction": {
+              "valEnum": ["concat"]
+            }
+          },
+          "then": {
+            "enum": ["string"]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+*Note: the `valEnum` in the `if` clause is comparing the actual literal value of the property, not the attribute type, as opposed to elsewhere in the attribute type schema, so we used a custom keyword `valEnum` instead of `enum` here.*
+
+### Operators with Type Rules
+
+The following operators will have type suggestion feature. 
+
+- Regular Expression
+- Dictionary Matcher
+- Keyword Search
+- Sentiment Analysis
+- Interval Join
+- Hash Join
+- Linear Regression
+- Sort Partitions
+- Aggregate
+- Unnest String
+- Type Casting
+- Word Cloud
+- ScatterPlot
+
+There may be more operators to be added in the future that will support type suggestion.
+
```

Comment:
I suggest adding links to PRs as references for this blog. similar to https://github.com/Texera/texera.github.io/commit/c2275c86aee8ed2ae591c5a8629877d61acc160e